src/index.cpp

```diff
@@ -1971,13 +2002,30 @@ template <typename IdType>
 std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search(const T *query, const size_t K, const uint32_t L,
                                                              IdType *indices, float *distances)
 {
+    IndexSearchContext<LabelT> context;
+    return search(query, K, L, indices, distances, context);
+}
+
+template <typename T, typename TagT, typename LabelT>
+template <typename IdType>
+std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search(const T *query, const size_t K, const uint32_t L,
```

Comment:
An after thought. Can we change the return type of search to return an IndexSearchContext object and return the stats, the number of results returned, timing information and what ever else we need via the return object?